File: 9-PalindromeNumber/9-PalindromeNumber.java
// Last updated: 8/2/2026, 1:05:58 AM
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){return false;}
        String str=Integer.toString(x);
        int st=0;
        int end=str.length()-1;
        while(st<end){
            if(str.charAt(st)==str.charAt(end)){
                st++;
                end--;
            }
            else{return false;}
        }
        return true;
    }
}

File: 9-PalindromeNumber/9-PalindromeNumber.c
// Last updated: 8/2/2026, 1:06:04 AM
bool isPalindrome(int x) {
    char str1[100],str2[100];
    int l;
    sprintf(str1, "%d", x);
    strcpy(str2,str1);
    l=strlen(str1);
    int start=0,end=(l-1);
    while(start<end)
    {
        char temp = str1[start];
        str1[start] = str1[end];
        str1[end] = temp;
        start++;
        end--;
    }
    if(strcmp(str1,str2)==0)
        return 1;
    else
        return 0;
}